### ISSUE TITLE
Diagnostics not shown in file peek views

### ISSUE THREAD

**Issue by mjbvz** [microsoft/typescript-go#2366]
1. Create two files:

```ts
// abc.ts
/**
 * @deprecated
 */
export function abc() { }
abc();
```

```ts
// other.ts
import { abc } from "./abc";

abc()
xyz;
```

2. With only `abc.ts` open, run `peek references` on `abc` (you can also use the references codelens for this)

**Bug**
In the peek view, no diagnostics are shown

<img width="894" height="484" alt="Image" src="https://github.com/user-attachments/assets/a3fc16f2-a80f-474c-a13c-5952e49a361d" />

They are shown with TS 6.0

**Comment by jakebailey** [comment]
We don't control this; is VS Code / the LSP client not opening that file and asking for diagnostics?

**Comment by mjbvz** [comment]
Yes maybe an lsp issue? cc @dbaeumer

**Comment by dbaeumer** [comment]
VS Code and LSP issue :-(. I assume the new TS extension uses pull diagnostics. We pull for open editor tabs and I guess that VS Code is not representing editors in peek views in the tab model. @lramos15 is there a way to know about these peek editor tabs?

**Comment by jakebailey** [comment]
Yes, we currently exclusively use pull diagnostic, which was how tsserver worked before, too. The TS extension then made the calls to ask for errors, so perhaps it already solved this issue once before?

**Comment by lramos15** [comment]
> @lramos15 is there a way to know about these peek editor tabs?

I'm not sure maybe @bpasero knows. I don't think the tabs API is the right place to represent this, unless we have that be a member property on a tab since we want tabs to represent logical UI tabs. Maybe the text document API is better here?

**Comment by bpasero** [comment]
@jrieken might have guidance

**Comment by jrieken** [comment]
Peek editors aren't tab-editors and I think the visible editors route needs to be taken here.

**Comment by dbaeumer** [comment]
Fixed in https://github.com/microsoft/vscode-languageserver-node/pull/1703

Will make a new next release so you can verify it.

**Comment by dbaeumer** [comment]
Server: 10.0.0-next.16
Client: 10.0.0-next.19
